File: Autos-stock-manager-services/src/main/java/org/autostock/dtos/DocumentRichDto.java
package org.autostock.dtos;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
public class DocumentRichDto {
    private Long id;
    private String type;
    private String typeLabel;
    private String nomFichier;
    private String description;
    private BigDecimal montant;
    private LocalDateTime dateUpload;
    private boolean principale;
    private String urlFichier;

    // Voiture
    private Long voitureId;
    private String voitureLabel;   // "Marque Modele Annee"
    private String couleur;
    private String vin;

    // Vendeur (depuis la vente associée à la voiture, si existe)
    private String vendeurNom;
    private String vendeurEmail;

    // Client (depuis la vente, si existe)
    private String clientNom;
    private String clientTelephone;
}

File: autos-stock-client/src/app/models/DocumentRichDto.ts
export interface DocumentRichDto {
  id: number;
  type: string;
  typeLabel: string;
  nomFichier?: string;
  description?: string;
  montant?: number;
  dateUpload?: string;
  principale?: boolean;
  urlFichier?: string;

  // Voiture
  voitureId?: number;
  voitureLabel?: string;
  couleur?: string;
  vin?: string;

  // Vendeur
  vendeurNom?: string;
  vendeurEmail?: string;

  // Client
  clientNom?: string;
  clientTelephone?: string;
}
